Stripe/STPPaymentCardTextField.m

```diff
@@ -1287,50 +1303,72 @@ - (NSAttributedString *)formTextField:(STPFormTextField *)formTextField
 
 - (void)formTextFieldTextDidChange:(STPFormTextField *)formTextField {
     STPCardFieldType fieldType = formTextField.tag;
-    if (fieldType == STPCardFieldTypeNumber) {
-        [self updateImageForFieldType:fieldType];
-        [self updateCVCPlaceholder];
-        // Changing the card number field can invalidate the cvc, e.g. going from 4 digit Amex cvc to 3 digit Visa
-        self.cvcField.validText = [self.viewModel validationStateForField:STPCardFieldTypeCVC] != STPCardValidationStateInvalid;
-    }
     
-    STPCardValidationState state = [self.viewModel validationStateForField:fieldType];
     formTextField.validText = YES;
-    switch (state) {
-        case STPCardValidationStateInvalid:
-            formTextField.validText = NO;
+
+    switch (fieldType) {
+
+        case STPCardFieldTypeNumber: {
+            NSString *number = [self.viewModel.cardNumber copy];
+            
+            // Changing the card number field can invalidate the cvc, e.g. going from 4 digit Amex cvc to 3 digit Visa
+            // it is not expected that the brand will change based on network response so we update this immediately
+            // as well as in the completion just in case
```

Comment:
We don't update brand in the completion block - should we? Or should we remove the last comment?

Reply:
brand is a computed property on the viewmodel so it's implicitly updated when we make the call to update the cvc placeholder again